src/services/StructuralMetadataUtils.js

```diff
@@ -97,7 +94,7 @@ export default class StructuralMetadataUtils {
           let endTime = convertToHHmmss(end);
           item.begin = this.toHHmmss(beginTime);
           if (beginTime > endTime) {
-            item.end = this.toHHmmss(fileLength);
+            item.end = this.toHHmmss(duration);
```

Comment:
@cjcolvar As long as this line doesn't break anything...everything else looks good to me provided it all runs fine?

Reply:
I didn't understand what was going on in setting `fileLength` (`Math.round(duration / 10) / 100`) so I just used duration which should be formatted correctly by `toHHmmss`.  What do you think?

Reply:
Yea makes sense.  I'd say go for it, and merge whenever you like.